spark/src/test/java/org/apache/iceberg/spark/source/TestDataSourceOptions.java

```diff
@@ -201,6 +201,36 @@ public void testSplitOptionsOverridesTableProperties() throws IOException {
     Assert.assertEquals("Spark partitions should match", 2, resultDf.javaRDD().getNumPartitions());
   }
 
+  @Test
+  public void testSplitOptionsOverridesTablePropertiesWithWriterLength() throws IOException {
+    String tableLocation = temp.newFolder("iceberg-table").toString();
+
+    HadoopTables tables = new HadoopTables(CONF);
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+    Map<String, String> options = Maps.newHashMap();
+    options.put(TableProperties.SPLIT_SIZE, String.valueOf(128L * 1024 * 1024)); // 128Mb
+    tables.create(SCHEMA, spec, options, tableLocation);
+
+    List<SimpleRecord> expectedRecords = Lists.newArrayList(
+        new SimpleRecord(1, "a"),
+        new SimpleRecord(2, "b")
+    );
+    Dataset<Row> originalDf = spark.createDataFrame(expectedRecords, SimpleRecord.class);
+    originalDf.select("id", "data").write()
+        .format("iceberg")
+        .mode("append")
+        .option("use-writer-length-as-file-size", true)
+        .save(tableLocation);
+
+    Dataset<Row> resultDf = spark.read()
+        .format("iceberg")
+        .option("split-size", String.valueOf(611 + 103)) // 611 bytes is the size of SimpleRecord(1,"a")
```

Comment:
@rdblue additional 103 in writer length, can this be bug in writer factory which returns buffer size after flush?
( no rush of merging this PR , I am trying to make sure changes are ok)

Reply:
Does this happen for all formats, or just one?

Reply:
Do we know why this is happening? I would expect Parquet to return the correct size after close. We should find out what's going on.

Reply:
I think this happens only for parquet. 
https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java#L142 
`writeStore` seems to return non zero results for `getBufferedSize` after close.